core/system/src/unvic.c

```diff
@@ -604,4 +561,19 @@ void unvic_init(void)
      * by IRQ number. For example, IRQ 0 has precedence over IRQ 1 if both have
      * the same priority level. */
     NVIC_SetPriorityGrouping(0);
+
+    /* Make sure the UNVIC_VECTOR_UNUSED is definitely NOT a valid box id. */
+    if (vmpu_is_box_id_valid(UNVIC_VECTOR_UNUSED)) {
+        HALT_ERROR(SANITY_CHECK_FAILED,
+            "UNVIC_VECTOR_UNUSED (%u) must not be a valid box id!\n",
+            UNVIC_VECTOR_UNUSED);
+    }
+
+    /* Copy the user interrupt table into the handlers. */
+    for (uint32_t ii=0; ii < NVIC_VECTORS; ii++) {
+        g_unvic_vector[ii].id = UNVIC_VECTOR_UNUSED;
+        g_unvic_vector[ii].hdlr = (TIsrVector) user_vtor[ii + NVIC_OFFSET];
+        /* set default priority (SVC must always be higher) */
+        NVIC_SetPriority(ii, __UVISOR_NVIC_MIN_PRIORITY);
```

Comment:
Is registering the user interrupt table interrupts with the public box really necessary? It sounds like a nice thing to do at first glance, but given that interrupts must be enabled before use, isn't just having enable (and the other `vIRQ` functions) register for the interrupt good enough? Is there some case I'm missing where an interrupt in the user interrupt table can be used without calling one of the `NVIC` or `vIRQ` functions?

If it is possible for a user to use an interrupt without calling any NVIC (or vIRQ) functions, how easy is it for a user to change the user interrupt table? Does a user need to change mbed OS code?

On the surface, I agree with this change. Interrupts in the user interrupt table are "legacy" interrupts and should be removed in favor of using the "non-legacy" vIRQ API. This fits with the uVisor story well of: the public box is for backwards compatibility; to do things with secure boxes you need to use uVisor APIs. However, if it is too hard to remove some interrupt from the user interrupt table (and I'd argue that having to change a local copy of mbed OS is too hard), then I think usability goes out the window if there is no other way to unregister the interrupt from the public box. We may need to rethink including the ability for a box to unregister interrupts.